queue_services/entity-emailer/tests/unit/test_tracker.py

```diff
@@ -429,12 +429,19 @@ async def test_should_correctly_track_retries_for_failed_processing(tracker_app,
     mock_msg = create_mock_message(message_payload)
 
     # mock out process_email function to throw exception to simulate failed scenario
-    with patch.object(worker, 'process_email', side_effect=QueueException('Queue Error.')):
-        for x in range(5):
+    with patch.object(worker, 'process_email', side_effect=EmailException('Queue Error.')):
+        for _ in range(6):
+            with pytest.raises(EmailException):
+                await worker.cb_subscription_handler(mock_msg)
+
+    # mock out process_email function not to process since message reaches max retries
+    with patch.object(worker, 'process_email', side_effect=EmailException('Queue Error.')):
+        for _ in range(5):
             await worker.cb_subscription_handler(mock_msg)
```

Comment:
Maybe we should just parametrize this test and test all the exception types(`OperationalError`, `EmailException`, `QueueException`, `Exception`) and verify it stops retrying after the retry limit is exceeded.